exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/metric_exporter/__init__.py

```diff
@@ -0,0 +1,449 @@
+            )
+        )
+        self._compression = compression or _compression_from_env()
+        self._session = session or requests.Session()
+        self._session.headers.update(self._headers)
+        self._session.headers.update(
+            {"Content-Type": "application/x-protobuf"}
+        )
+        if self._compression is not Compression.NoCompression:
+            self._session.headers.update(
+                {"Content-Encoding": self._compression.value}
+            )
+
+        instrument_class_temporality = {}
+        if (
+                environ.get(
+                    OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE,
+                    "CUMULATIVE",
+                )
+                        .upper()
+                        .strip()
+                == "DELTA"
+        ):
+            instrument_class_temporality = {
+                Counter: AggregationTemporality.DELTA,
+                UpDownCounter: AggregationTemporality.CUMULATIVE,
+                Histogram: AggregationTemporality.DELTA,
+                ObservableCounter: AggregationTemporality.DELTA,
+                ObservableUpDownCounter: AggregationTemporality.CUMULATIVE,
+                ObservableGauge: AggregationTemporality.CUMULATIVE,
+            }
+        else:
+            instrument_class_temporality = {
+                Counter: AggregationTemporality.CUMULATIVE,
+                UpDownCounter: AggregationTemporality.CUMULATIVE,
+                Histogram: AggregationTemporality.CUMULATIVE,
+                ObservableCounter: AggregationTemporality.CUMULATIVE,
+                ObservableUpDownCounter: AggregationTemporality.CUMULATIVE,
+                ObservableGauge: AggregationTemporality.CUMULATIVE,
+            }
+        instrument_class_temporality.update(preferred_temporality or {})
+
+        MetricExporter.__init__(
+            self,
+            preferred_temporality=instrument_class_temporality,
+            preferred_aggregation=preferred_aggregation,
+        )
+
+    def _export(self, serialized_data: str):
+        data = serialized_data
+        if self._compression == Compression.Gzip:
+            gzip_data = BytesIO()
+            with gzip.GzipFile(fileobj=gzip_data, mode="w") as gzip_stream:
+                gzip_stream.write(serialized_data)
+            data = gzip_data.getvalue()
+        elif self._compression == Compression.Deflate:
+            data = zlib.compress(bytes(serialized_data))
+
+        return self._session.post(
+            url=self._endpoint,
+            data=data,
+            verify=self._certificate_file,
+            timeout=self._timeout,
+        )
+
+    @staticmethod
+    def _retryable(resp: requests.Response) -> bool:
+        if resp.status_code == 408:
+            return True
+        if resp.status_code >= 500 and resp.status_code <= 599:
+            return True
+        return False
+
+    def _translate_data(
```

Comment:
Better to move this method in a mixin if this is the same code as in the gRPC exporter.

Reply:
@srikanthccv this won't address the original intent of not duplicating the `_translate_data` function body in both the HTTP and GRPC exporters.

Reply:
It's not possible to share code from gRPC exporter without another common package `*-otlp-exporter-common` which is out of scope for this but what is doable is avoid the code repetition in the HTTP exporter and that's where the mixin and the rest of the definitions like `_translate_value`,`get_resource_data`, `compression_from_env` can follow the similar structure from gRPC exporter.

Reply:
Created tracking issue for this: https://github.com/open-telemetry/opentelemetry-python/issues/2990.

Let's not block this pr.